diff --git a/node_modules/eslint-plugin-react/lib/rules/jsx-filename-extension.js b/node_modules/eslint-plugin-react/lib/rules/jsx-filename-extension.js
index 4a2cfee..3c7e1c0 100644
--- a/node_modules/eslint-plugin-react/lib/rules/jsx-filename-extension.js
+++ b/node_modules/eslint-plugin-react/lib/rules/jsx-filename-extension.js
@@ -61,7 +61,9 @@ module.exports = {
   },
 
   create(context) {
-    const filename = context.getFilename();
+    const filename = typeof context.getFilename === 'function'
+      ? context.getFilename()
+      : context.filename;
 
     let jsxNode;
 
diff --git a/node_modules/eslint-plugin-react/lib/util/version.js b/node_modules/eslint-plugin-react/lib/util/version.js
index c9dc683..3feda5c 100644
--- a/node_modules/eslint-plugin-react/lib/util/version.js
+++ b/node_modules/eslint-plugin-react/lib/util/version.js
@@ -28,7 +28,11 @@ function resetDetectedVersion() {
 
 function resolveBasedir(contextOrFilename) {
   if (contextOrFilename) {
-    const filename = typeof contextOrFilename === 'string' ? contextOrFilename : contextOrFilename.getFilename();
+    const filename = typeof contextOrFilename === 'string'
+      ? contextOrFilename
+      : (typeof contextOrFilename.getFilename === 'function'
+        ? contextOrFilename.getFilename()
+        : contextOrFilename.filename);
     const dirname = path.dirname(filename);
     try {
       if (fs.statSync(filename).isFile()) {
